lib/interfaces/event-request-object.interface.ts

```diff
@@ -0,0 +1,3 @@
+export type EventRequestObject<T> = {
```

Comment:
LGTM overall, I'd just change this to 
```suggestion
export type EventPayloadHost<T> = {
```